File: Mathématiques/3 – Arithmétique/4 - Étude des nombres premiers.py
nombre = int(input())
prime = []
for i in range(2, nombre+1, 1):
    isPrime = True
    for n in prime:
        if i % n == 0:
            isPrime = False
    if isPrime:
        prime.append(i)

if prime[len(prime) - 1] == nombre:
    print("Premier")
else:
    print("Composé")

File: Mathématiques/3 – Arithmétique/4 - Étude des nombres premiers.cpp
#include <iostream>
using namespace std;

int generatePrimeNumber(int n)
{
  int length = 1;
  int primeNumber[n] = {2};
  int number = 2;
  while (number <= n)
  {
    bool isPrime = true;
    for (int i = 0; i < length; i++)
    {
      if (number % primeNumber[i] == 0)
      {
        isPrime = false;
        break;
      }
    }
    if (isPrime)
    {
      primeNumber[length] = number;
      length++;
    }
    number++;
  }
  return primeNumber[length - 1];
}

int main()
{
  int n;
  cin >> n;
  if (n == generatePrimeNumber(n))
  {
    cout << "Premier";
  }
  else
  {
    cout << "Composé";
  }
}